feos-core/src/state/mod.rs

```diff
@@ -693,7 +694,20 @@ impl<U: EosUnit, E: EquationOfState> State<U, E> {
     }
 
     /// Creates a [StateHD] taking the first and second (partial) derivatives.
-    pub fn derive2(
+    pub fn derive2(&self, derivative: Derivative) -> StateHD<Dual2_64> {
+        let mut t = Dual2_64::from(self.reduced_temperature);
+        let mut v = Dual2_64::from(self.reduced_volume);
+        let mut n = self.reduced_moles.mapv(Dual2_64::from);
+        match derivative {
+            Derivative::DT => t = t.derive(),
```

Comment:
maybe change the other `derive` functions to also use the `derive` method of the respective dual numbers?